Loop/Managers/TestingScenariosManager.swift

```diff
@@ -224,71 +224,73 @@ extension TestingScenariosManager {
     }
 
     private func loadScenario(_ scenario: TestingScenario, completion: @escaping (Error?) -> Void) {
-        guard FeatureFlags.scenariosEnabled else {
-            fatalError("\(#function) should be invoked only when scenarios are enabled")
-        }
-
         func bail(with error: Error) {
             activeScenarioURL = nil
             log.error("%{public}@", String(describing: error))
             completion(error)
         }
         
-        let instance = scenario.instantiate()
-        
-        var testingCGMManager: TestingCGMManager?
-        var testingPumpManager: TestingPumpManager?
-        
-        if instance.hasCGMData {
-            if let cgmManager = deviceManager.cgmManager as? TestingCGMManager {
-                if instance.shouldReloadManager?.cgm == true {
-                    testingCGMManager = reloadCGMManager(withIdentifier: cgmManager.pluginIdentifier)
+        Task {
+            guard FeatureFlags.scenariosEnabled else {
+                fatalError("\(#function) should be invoked only when scenarios are enabled")
+            }
+            
+            let instance = scenario.instantiate()
+            
+            var testingCGMManager: TestingCGMManager?
+            var testingPumpManager: TestingPumpManager?
+            
+            if instance.hasCGMData {
+                if let cgmManager = deviceManager.cgmManager as? TestingCGMManager {
+                    if instance.shouldReloadManager?.cgm == true {
+                            testingCGMManager = await reloadCGMManager(withIdentifier: cgmManager.pluginIdentifier)
+                    } else {
+                        testingCGMManager = cgmManager
+                    }
                 } else {
-                    testingCGMManager = cgmManager
+                    bail(with: ScenarioLoadingError.noTestingCGMManagerEnabled)
+                    return
                 }
-            } else {
-                bail(with: ScenarioLoadingError.noTestingCGMManagerEnabled)
-                return
             }
-        }
-        
-        if instance.hasPumpData {
-            if let pumpManager = deviceManager.pumpManager as? TestingPumpManager {
-                if instance.shouldReloadManager?.pump == true {
-                    testingPumpManager = reloadPumpManager(withIdentifier: pumpManager.pluginIdentifier)
+            
+            if instance.hasPumpData {
+                if let pumpManager = deviceManager.pumpManager as? TestingPumpManager {
+                    if instance.shouldReloadManager?.pump == true {
+                        testingPumpManager = reloadPumpManager(withIdentifier: pumpManager.pluginIdentifier)
+                    } else {
```

Comment:
@Camji55 should this also be updated to use `prepareForDeactivation`. Don't really want to stir the pot, but this also seems like a good time to do it.

Reply:
Not too sure what the `prepareForDeactivation` does. I'll defer to you on that! Happy to make the change if the lift is light!